lib/ezpool.rb

```diff
@@ -128,6 +133,7 @@ def checkout(options = {})
     @mutex.synchronize do
       @checked_out_connections[conn_wrapper.raw_conn.object_id] = conn_wrapper
     end
+    conn_wrapper.touch_connection
```

Comment:
Should we `touch` on `checkin` too? If GDS operation takes long time, which it can, we will expire it sooner otherwise

Reply:
I was thinking about this. There is a tradeoff between these options.

- Touch only on the checkout: more frequent reconnections than it could be in case we have many operations inside `with` block.
- Touch both on the checkout and the checkin: connections could miss reconnection and become stale in case we have long-running operations and additional non-gds code inside `with` block
I'm not sure how exactly gds determine if connection is stale or not (client request time or server response). However, I just feel like more frequent reconnections is more reliable way.